### PR TITLE
'client' param for validate & invalidate function

### PR DESCRIPTION
**Pull request by InventivetalentDev**
This adds the `clientToken` parameter to the [`validate` function](http://wiki.vg/Authentication#Validate) and the [`invalidate` function](http://wiki.vg/Authentication#Invalidate)

### CODE REVIEW

**Comment by Foorack** [comment]
@zekesonxx Any update on this? :smiley:

**Comment by rom1504** [comment]
@wvffle what do you think about this one ?

**Comment by wvffle** [comment]
LGTM

**Comment by rom1504** [comment]
it needs rebasing

**Comment by IdanHo** [comment]
this param should be optional, shouldnt it?

**Comment by InventivetalentDev** [comment]
@IdanHo Yes, it is.  I guess I could just change it to be another function allowing the client id instead of replacing the existing one.

File: test/index.js [hunk 1]
```diff
@@ -161,6 +161,16 @@ describe('Yggdrasil', function () {
         done()
       })
     })
+    it('should return undefined on valid response', function (done) {
+      cscope.post('/validate', {
+        accessToken: 'a magical key',
+        clientToken: 'bacon'
+      }).reply(200)
+      ygg.validateClient('a magical key', 'bacon', function (err) {
+        should(err).be.undefined
```
Comment (wvffle) [inline test/index.js]: This should be a function call. `/* ... */.undefined()`

File: lib/Client.js [hunk 2]
```diff
@@ -70,6 +70,38 @@ Client.validate = function (token, cb) {
...
+ */
+Client.validateClient = function (token, client, cb) {
+  var host = this.host || defaultHost
+  utils.call(host, 'validate', {
+    accessToken: token,
+    clientToken: client
+  }, function (err, data) {
+    cb(err)
+  })
+}
```
Comment (wvffle) [inline lib/Client.js]: I there is a significant difference between `token` and `client` which is easily distinguishable then I propose to stay with one `.validate(clientOrAccessToken, cb)` method and detect which one is used.

Comment (InventivetalentDev) [reply lib/Client.js]: It does need both access and client token though, so you can't really just have a single param

Comment (wvffle) [reply lib/Client.js]: I understand, then `validateClient` is okay

File: lib/Client.js [hunk 3]
```diff
@@ -70,6 +70,38 @@ Client.validate = function (token, cb) {
...
+ */
+Client.invalidate = function (token, client, cb) {
+  var host = this.host || defaultHost
+  utils.call(host, 'invalidate', {
+    accessToken: token,
+    clientToken: client
+  }, function (err, data) {
+    cb(err)
+  })
+}
```
Comment (wvffle) [inline lib/Client.js]: Add `invalidateClient` for client

**Comment by IdanHo** [comment]
i think you commited to the wrong repo

**Comment by rom1504** [comment]
please reopen a new clean PR if needed

**Pull request closed without merging**